src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -244,32 +272,56 @@ const SKUpdate& SKNMEA2000Parser::parse128259(const SKSourceInput& input, const
 
 // ****************************************************************************
 //  PGN 128267  Water depth
+//  Water depth relative to the transducer and offset of the measuring transducer.
+//  Water depth is either below water surface or below lowest point of vessel.
+//  Offset – If the value of this field is positive, it represents the
+//  difference in depth between the transducer and the waterline of the vessel.
+//  If the value of this field is negative, it represents the difference in
+//  depth between the transducer and the keel of the vessel.
 // ****************************************************************************
 const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
   unsigned char sid;
-  double depthBelowTransducer = N2kDoubleNA;
+  double depth = N2kDoubleNA;
   double offset = N2kDoubleNA;
 
-  if (ParseN2kWaterDepth(msg, sid, depthBelowTransducer, offset)) {
-    if (!N2kIsNA(depthBelowTransducer)) {
-      SKUpdateStatic<3> *update = new SKUpdateStatic<3>();
+  if (ParseN2kWaterDepth(msg, sid, depth, offset)) {
+    if (!N2kIsNA(depth)) {
+      SKUpdateStatic<5> *update = new SKUpdateStatic<5>();
       update->setTimestamp(timestamp);
 
       SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
       update->setSource(source);
+      update->setEnvironmentDepthBelowTransducer(depth);
 
-      update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
-
-      // When offset is negative, it's the distance between transducer and keel
       if (!N2kIsNA(offset)) {
-        if (offset < 0) {
+        if (offset == 0) {
+          // When you have zero offset than the transducer is mounted on the
```

Comment:
Did you find a source for this information?

Reply:
The Maretron manual says:

> 3:  Offset – If the value of this field is positive, it represents the difference in depth between the transducer and the waterline of the vessel.  If the value of this field is negative, it represents the difference in depth between the transducer and the keel of the vessel.  The unit of measurement for this field is 1x10-3 m.

I think when the value is 0 we do not know where it is.

Reply:
Also the Raymarine manual says:
> If an offset is not applied, displayed depth readings represent the distance from the transducer to the sea bed

Reply:
A mayor difference to this we had before is, that offset = zero is a defined state 

> I think when the value is 0 we do not know where it is.

Generally spoken: you never know (by the instruments) the depth from the keel to the ground, when set to "depth from water surface" and on opposite you do not know where the transducer is mounted if set to "depth below keel".

**For offset=0**
Think of a boat, where the transducer **is** at the lowest point.
Then you would enter zero to offset if you want to display depth below keel, no?
If an offset is zero, than **we do know that someone had set it on purpose.** 
(Otherwise it would be NaN)

If someone does not enter any offset, then we know that the device was not set up (correctly). So we just can say it is the depth below sensor and not more.
This was and is still the way the code is doing it.
I think that this is meant with "not applied" (as zero is an active setting)

But may be I am totally wrong and the offset is Zero on a brand new unit.
We should ask somebody from Raymarine, Maretron or Airmar if they make a difference between "not set" / "not applied" and defined 0......

And what will then  actually be displayed in the two different settings **depth below keel** and **depth below water surface**???

Reply:
> Think of a boat, where the transducer is at the lowest point.

Yes but think of a boat where the transducer is right at the floatation point. 😛 

I think they gave up on these two situations when they decided to use a positive/negative value in the offset to indicate the reference. They just do not support it and I think we should not support it either.

I think in practice, you would say `+0.01` if it's right below the keel and `-0.01` if it's right below the surface. Saying 0 is like saying no offset and in that case we do not know where the sensor is located.

> But may be I am totally wrong and the offset is Zero on a brand new unit.

Yes it would be useful to know what a new device sends.

> And what will then actually be displayed in the two different settings depth below keel and depth below water surface???

If we do not have the information we just do not send it in SignalK. Displays should not show it either. That is a very common thing. There are lots of possible data on the network and it's not uncommon to be missing data.